helios-services/src/main/java/com/spotify/helios/servicescommon/statistics/FastForwardReporter.java

```diff
@@ -0,0 +1,218 @@
+import com.codahale.metrics.Metered;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Snapshot;
+import com.codahale.metrics.Timer;
+
+import eu.toolchain.ffwd.FastForward;
+import eu.toolchain.ffwd.Metric;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+
+import io.dropwizard.lifecycle.Managed;
+
+/**
+ * Sends the metrics in a v3 MetricRegistry to
+ * <a href="https://github.com/spotify/ffwd">FastForward</a>.
+ *
+ * <p>The String "name" of the Metric is sent as the 'what' attribute in the FastForward metric. No
+ * translation of the (possibly long, Java classname-ish) name is done. The type of metric is
+ * included in the 'metric_type' attribute.</p>
+ *
+ * <p>For meters, timers, and histograms, the various component values of the
+ * com.codahale.metrics.Metric (m1/m5/m15, p50, p99 etc) are sent as individual {@link Metric}
+ * instances. The 'stat' attribute is filled out with the corresponding statistic. </p>
+ *
+ * <p>Note that not every statistic from Meters/Timers/Histograms is sent to FastForward in order
+ * to limit the amount of data being sent and needing to be stored downstream. For Metered
+ * instances, only 1m and 5m is sent. For Histograms, just median, p75 and p99 (plus
+ * min/max/mean/stddev).
+ * </p>
+ */
+public class FastForwardReporter implements Managed {
+
+  private static final Logger log = LoggerFactory.getLogger(FastForwardReporter.class);
+
+  public static FastForwardReporter create(MetricRegistry registry, Optional<HostAndPort> address,
+                                           String metricKey, int intervalSeconds)
+      throws SocketException, UnknownHostException {
+
+    final FastForward ff;
+    if (address.isPresent()) {
+      ff = FastForward.setup(address.get().getHostText(), address.get().getPort());
+    } else {
+      ff = FastForward.setup();
+    }
+
+    final ThreadFactory threadFactory = new ThreadFactoryBuilder()
+        .setDaemon(true)
+        .setNameFormat("fast-forward-reporter-%d")
+        .build();
+
+    final ScheduledExecutorService executorService =
+        Executors.newSingleThreadScheduledExecutor(threadFactory);
+
+    return new FastForwardReporter(ff, registry, executorService, metricKey, intervalSeconds,
+                                   TimeUnit.SECONDS);
+  }
+
+  private final FastForward fastForward;
+  private final MetricRegistry metricRegistry;
+  private final ScheduledExecutorService executor;
+  private final String key;
+  private final long interval;
+  private final TimeUnit intervalTimeUnit;
+
+  public FastForwardReporter(FastForward fastForward, MetricRegistry metricRegistry,
```

Comment:
It compiles for me. The only thing that directly calls this constructor is the test. MasterService and AgentService just use the public static ctor above.